andela/mervels-bn-backend
# #168131407 Integrate coveralls code coverage service

#### What does this PR do?
Adds the coveralls code coverage services with a coverall badge.
#### Description of Task to be completed?
- Add the repo to the coveralls 
- Install required dependencies `nyc` and `coveralls`
- Add script to create the coverage
- Add a script in travis.yml to send coverage to coveralls.io
#### How should this be manually tested?
- Clone this repo and checkout this branch.
- Run `nmp run cover` which generate coverage folder with coverage info.
- You can also run `npm run test` generates a detailed table of test coverage.
- Check out the readme file for the badge.
#### Any background context you want to provide?
This same PR was used to add a pre-commit hook 
- Install `husky` and `lint-staged` to create the hook.
It was also used to add prettier
- Install `prettier-eslint` 
- Run `npm run lint:fix` for prettier to edit the files.
#### What are the relevant pivotal tracker stories?
[168131407](https://www.pivotaltracker.com/story/show/168131407)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A
